taker-electron/src/main/main.ts

```diff
@@ -132,34 +134,64 @@ app.on("window-all-closed", () => {
     }
 });
 
-app
-    .whenReady()
-    .then(async () => {
-        await createWindow();
+// retry checking random port by the given amount of max retries.
+const retry = (maxRetries: number, fn: (port: number) => Promise<number>, port: number): Promise<number> => {
+    const minPort = 10_000;
+    const maxPort = 65_535;
 
-        log.debug("Waiting for ItchySats to become available.");
+    return fn(port).catch(function(err) {
+        log.info(`Port: ${port} is not available retrying another random port. Retries: ${maxRetries}`);
+        if (maxRetries <= 0) {
+            log.error(`Reached max amount of retries. quitting.`);
+            throw err;
+        }
+        return retry(maxRetries - 1, fn, Math.floor(Math.random() * (maxPort - minPort + 1) + minPort));
+    });
+};
 
-        process.env.ITCHYSATS_ENV = "electron";
+// checks if the provided port is already taken on the localhost.
+const checkAvailablePort = (port: number): Promise<number> =>
+    new Promise((resolve, reject) => {
+        const server = nodenet.createServer();
+        server.unref();
+        server.on("error", reject);
+        log.debug(`Trying port: ${port}`);
+        server.listen({ port, host: "127.0.0.1" }, () => {
+            const { port } = <any> server.address();
+            server.close(() => {
+                log.debug(`Found open port: ${port}!`);
+                resolve(port);
+            });
+        });
+    });
```

Comment:
nit: it would be good to extract this into a function to hide the implementation details